### PR TITLE
Feat/savings details flyout

### PR DESCRIPTION
**Pull request by jjBlockchain**

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/data/components/interest/sagas.ts [hunk 1]
```diff
@@ -66,16 +66,14 @@ export default ({ api }: { api: APIType }) => {
...
-      case 'BTC':
-        const accountsR = yield select(
-          selectors.core.common.btc.getAccountsBalances
-        )
-        const defaultIndex = yield select(
-          selectors.core.wallet.getDefaultAccountIndex
-        )
-        defaultAccountR = accountsR.map(nth(defaultIndex))
-        break
+    if (payload.coin === 'BTC') {
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/data/components/interest/sagas.ts:69]: Why did you change this from a switch statement?

File: packages/blockchain-wallet-v4-frontend/src/data/components/interest/types.ts [hunk 2]
```diff
@@ -18,16 +18,20 @@ export type InterestFormValuesType = {
 }
 
 export enum InterestSteps {
+  'DETAILS',
   'DEPOSIT'
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/data/components/interest/types.ts:22]: Should details come after DEPOSIT?

Comment (plondon) [reply packages/blockchain-wallet-v4-frontend/src/data/components/interest/types.ts:22]: enum evaluates to something like:

steps = {
  'DETAILS': 0,
  'DEPOSIT': 1
}

Then later we can use the indices to determine if the modal should go to the left or the right direction.

File: packages/blockchain-wallet-v4-frontend/src/modals/Interest/InterestDetails/Summary/model.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+export const getNextMonth = (): string => {
+  const current = new Date()
+  const next =
+    current.getMonth() === 11
+      ? new Date(current.getFullYear() + 1, 0, 1)
+      : new Date(current.getFullYear(), current.getMonth() + 1, 1)
+
+  return (
+    months[next.getMonth()] + ' ' + next.getDate() + ', ' + next.getFullYear()
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/modals/Interest/InterestDetails/Summary/model.ts:24]: Can you use the `moment` library here instead? I think you could do just `moment.add(1, 'months')`.

https://momentjs.com/ (we already have the lib installed)

File: packages/blockchain-wallet-v4-frontend/src/modals/Interest/InterestDetails/index.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    return data.cata({
+      Success: val => (
+        <Success
+          {...val}
+          {...this.props}
+          handleDepositClick={this.handleDepositClick}
+          handleSimpleBuyClick={this.handleSimpleBuyClick}
+        />
+      ),
+      Failure: () => <DataError onClick={this.handleRefresh} />,
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/modals/Interest/InterestDetails/index.tsx:59]: If the details page fails to load refresh should take users to the DEPOSIT step? Or should we try to refetch w/e info failed to load?

File: packages/blockchain-wallet-v4-frontend/src/modals/Interest/InterestDetails/index.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+  interestActions: bindActionCreators(actions.components.interest, dispatch),
+  simpleBuyActions: bindActionCreators(actions.components.simpleBuy, dispatch)
+})
+
+const enhance = compose(
+  connect(
+    mapStateToProps,
+    mapDispatchToProps
+  )
+)
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/modals/Interest/InterestDetails/index.tsx:80]: Please use the new types set up using `connector = ` and `ConnectedProps<typeof connector>`. Template here: https://github.com/blockchain/blockchain-wallet-v4-frontend/blob/c71e5d9595ec1ca69bb470fb07888980b9460ca2/packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/EnterAmount/index.tsx

File: packages/blockchain-wallet-v4-frontend/src/modals/Interest/index.tsx [hunk 6]
```diff
@@ -35,16 +55,28 @@ class Interest extends PureComponent<Props, State> {
...
       </Flyout>
     )
   }
 }
 
+const mapStateToProps = (state: RootState) => ({
+  step: selectors.components.interest.getStep(state)
+})
+
 const enhance = compose<any>(
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/modals/Interest/index.tsx:77]: Please move types to bottom of the file and use `connector`

File: packages/blockchain-wallet-v4-frontend/src/scenes/Interest/index.tsx [hunk 7]
```diff
@@ -39,6 +39,7 @@ type LinkStatePropsType = {
 }
 type LinkDispatchPropsType = {
   identityVerificationActions: typeof actions.components.identityVerification
+  interestActions: typeof actions.components.interest
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/scenes/Interest/index.tsx:42]: Please move types to bottom of the file and use `connector`

File: packages/blockchain-wallet-v4-frontend/src/data/components/interest/types.ts (hunk 2 above)
Comment (jjBlockchain) [reply packages/blockchain-wallet-v4-frontend/src/data/components/interest/types.ts:22]: Yes **DETAILS** should be the beginning because once user has deposited then details will be the first flyout they can see. I set up reducer to have **DEPOSIT** as default to deal with this edge case

File: packages/blockchain-wallet-v4-frontend/src/modals/Interest/InterestDetails/Summary/model.ts (hunk 3 above)
Comment (jjBlockchain) [reply packages/blockchain-wallet-v4-frontend/src/modals/Interest/InterestDetails/Summary/model.ts:24]: Ahh forgot about that library. I will add changes.

File: packages/blockchain-wallet-v4-frontend/src/data/components/interest/sagas.ts (hunk 1 above)
Comment (jjBlockchain) [reply packages/blockchain-wallet-v4-frontend/src/data/components/interest/sagas.ts:69]: Was hitting a TS error. I think it was because I didn't have a default. Should I change it back to a switch statement and just break on default?

Comment (plondon) [reply packages/blockchain-wallet-v4-frontend/src/data/components/interest/sagas.ts:69]: Yes please

File: packages/blockchain-wallet-v4-frontend/src/modals/Interest/InterestDetails/Summary/index.tsx [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+            id='modals.interest.summary.accrued'
+            defaultMessage='Accrued interest this month'
+          />
+          <TooltipHost id='modals.interest.summary.accrued.tooltip'>
+            <TooltipIcon name='info' size='12px' />
+          </TooltipHost>
+        </Text>
+        <Text color='grey600' size='14px' weight={500}>
+          {fiatToString({
+            value: 0.00001,
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/modals/Interest/InterestDetails/Summary/index.tsx:65]: Should this value be hardcoded?

File: packages/blockchain-wallet-v4-frontend/src/modals/Interest/InterestDetails/Summary/index.tsx [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+          {' - '}
+          <Link href='#' target='_blank' size='14px' weight={500}>
+            <FormattedMessage
+              id='modals.interest.summary.moredetails'
+              defaultMessage='More details'
+            />
+          </Link>
+        </Text>
+        <Text color='grey600' size='14px' weight={500}>
+          3.00%
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/modals/Interest/InterestDetails/Summary/index.tsx:105]: Should this value be hardcoded?

File: packages/blockchain-wallet-v4-frontend/src/modals/Interest/InterestDetails/template.success.tsx [hunk 10]
```diff
@@ -0,0 +1,198 @@
...
+            <Text color='grey600' size='14px' weight={500}>
+              <FormattedMessage
+                id='modals.interest.detailsbalance'
+                defaultMessage='Your {coin} Balance'
+                values={{ coin }}
+              />
+            </Text>
+            <Text color='grey800' size='20px' weight={600}>
+              {fiatToString({
+                value: 6726.43,
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/modals/Interest/InterestDetails/template.success.tsx:116]: Ok this _definitely_ shouldnt be hardcoded haha :)

**Comment by plondon** [comment]
@TheLeoB can you help JJ resolve these conflicts?

**Comment by TheLeoB** [comment]
> @TheLeoB can you help JJ resolve these conflicts?

Done.
